geomet/esri.py

```diff
@@ -0,0 +1,202 @@
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+import json
+#-------------------------------------------------------------------------
+def load(source_file):
+    """
+    Converts Esri Json File to GeoJSON.
+
+    :param source_file:
+        Path to a file that contains the Esri JSON data.
+    
+    :returns: 
+         A GeoJSON `dict` representing the geometry read from the file.
+
+    """
+    with open(source_file, 'r') as reader:
+        data = json.loads(reader.read())
+        return loads(data)
+#-------------------------------------------------------------------------
+def loads(string):
+    """
+    Construct a GeoJSON `dict` from Esri JSON (string/dict).
+
+    :param string:
+        The Esri JSON geometry representation
+
+    :returns: 
+         A GeoJSON `dict` representing the geometry read from the file.
+    """
+    if isinstance(string, str):
+        string = json.loads(string)
+    if 'rings' in string:
+        return _esri_to_geojson_convert['rings'](string)
+    elif 'paths' in string:
+        return _esri_to_geojson_convert['paths'](string)
+    elif 'x' in string or 'y' in string:
+        return _esri_to_geojson_convert['x'](string)
+    elif 'points' in string:
+        return _esri_to_geojson_convert['points'](string)
+    else:
+        raise ValueError("Invalid EsriJSON: %s" % string)
+#-------------------------------------------------------------------------
+def dump(obj, dest_file):
+    """
+    Converts GeoJSON to Esri JSON File.
+    """
+    with open(dest_file, 'w') as writer:
+        writer.write(json.dumps(dumps(obj)))
+    return dest_file
+#-------------------------------------------------------------------------
+def dumps(obj):
+    """
+    Dump a GeoJSON-like `dict` to a Esri JSON.
+    """
+    if 'type' in obj and \
+        obj['type'].lower() in _gj_to_esri.keys():
+        convert = _gj_to_esri[obj['type'].lower()]
+        return convert(obj)
+    else:
+        raise ValueError("Invalid GeoJSON type %s" % obj)
+#-------------------------------------------------------------------------
+def _load_geojson_point(obj):
+    """
+    Loads GeoJSON to Esri JSON for Geometry type Point.
+
+    """
+    coords = obj['coordinates'] 
+    return {'x' : coords[0], 'y' : coords[1], "spatialReference" : {'wkid' : 4326}}
+#-------------------------------------------------------------------------
+def _load_geojson_multipoint(obj):
+    """
+    Loads GeoJSON to Esri JSON for Geometry type MultiPoint.
+
+    """
+    coordkey = ([d for d in obj if d.lower() == 'coordinates']
```

Comment:
Since GeoJSON is a mishmash of programs, it allows 'cOOrdInatEs' to be caseless but accepted.  I think it makes geomet more flexible.

Reply:
That's nonsense. The GeoJSON RFC specifies "coordinates" lowercase. Flexible here means admitted non-conforming GeoJSON, and that's wrong.

Reply:
Since JSON has case-sensitive keys, and GeoJSON has clearly defined the key as `"coordinates"` in lower case, I too believe it should not be performing `tolower` here (or anywhere else, for that matter).

Reply:
Jeez ok, I'll remove it :)  Seems like it's the wild west out there for GeoJSON at times.

Reply:
I agree - GeoJSON from the wild can be a bit...chaotic....but the best way we can try to fix that is to force inputs and outputs to conform to the spec.